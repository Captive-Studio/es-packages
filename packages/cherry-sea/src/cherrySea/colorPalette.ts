import type { ColorPalette } from '../type.js';
import { aquaSerenity } from './colorPalette/aquaSerenity.js';
import { blueGrey } from './colorPalette/blueGrey.js';
import { pinkVivid } from './colorPalette/pinkVivid.js';
import { lightBlueVivid } from './colorPalette/lightBlueVivid.js';
import { limeGreen } from './colorPalette/limeGreen.js';
import { yellowVivid } from './colorPalette/yellowVivid.js';
import { redVivid } from './colorPalette/redVivid.js';
import { purple } from './colorPalette/purple.js';
import { goldenFizz } from './colorPalette/goldenFizz.js';

export const colorPalette = {
  aquaSerenity,
  blueGrey,
  pinkVivid,
  lightBlueVivid,
  limeGreen,
  yellowVivid,
  redVivid,
  goldenFizz,
  purple,
} satisfies Record<string, ColorPalette>;
