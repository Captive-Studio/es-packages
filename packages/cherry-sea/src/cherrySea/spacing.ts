import type { Spacing } from '../type.js';

export const spacing: Spacing = {
  0: '0px',
  25: '4px',
  50: '8px',
  75: '12px',
  100: '16px',
  150: '24px',
  200: '32px',
  300: '48px',
  400: '64px',
  600: '96px',
  800: '128px',
  1200: '192px',
  1600: '256px',
  2400: '384px',
  3200: '512px',
  4000: '640px',
  4800: '768px',
};
