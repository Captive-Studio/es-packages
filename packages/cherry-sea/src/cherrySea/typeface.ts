import { Typeface } from '../type.js';

const fallback = ['-apple-system', 'Segoe UI', 'roboto', 'noto sans', 'ubuntu', 'cantarell', 'helvetica neue'];

export const typeface = {
  brand: Typeface({
    fontFamily: ['Barlow', ...fallback],
  }),
  plain: Typeface({
    fontFamily: ['Inter', ...fallback],
  }),
};
