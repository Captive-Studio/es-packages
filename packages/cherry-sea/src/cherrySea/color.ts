import type { Color } from '../type.js';
import { colorPalette } from './colorPalette.js';

const white = '#fff';
const defaultError = 'red';

export const color = {
  // emphasisOnBlue: colorPalette.goldenFizz[400],

  // Primary
  primary: colorPalette.pinkVivid[500],
  onPrimary: white,
  primaryContainer: defaultError,
  onPrimaryContainer: defaultError,
  // Secondary
  secondary: defaultError,
  onSecondary: defaultError,
  secondaryContainer: defaultError,
  onSecondaryContainer: defaultError,
  // Tertiary
  tertiary: defaultError,
  onTertiary: defaultError,
  tertiaryContainer: defaultError,
  onTertiaryContainer: defaultError,
  // Error
  error: defaultError,
  onError: defaultError,
  errorContainer: defaultError,
  onErrorContainer: defaultError,

  surface: colorPalette.blueGrey[50],
  onSurface: colorPalette.blueGrey[900],

  outlineVariant: defaultError,
  onSurfaceVariant: colorPalette.blueGrey[700],

  outline: defaultError,
  white,
} satisfies Color;
