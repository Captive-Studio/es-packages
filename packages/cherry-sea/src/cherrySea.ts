import type { Theme } from './type.js';
import { spacing } from './cherrySea/spacing.js';
import { typeface } from './cherrySea/typeface.js';

/**
 * Cherry sea theme configuration
 */
export const cherrySea = {
  typeface,
  spacing,
} satisfies Theme;
