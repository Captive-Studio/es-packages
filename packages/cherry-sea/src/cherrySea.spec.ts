import { describe, it, expect } from 'vitest';
import { cherrySea } from './cherrySea.js';

describe('cherrySea', () => {
  it('matches snapshot', () => {
    expect(cherrySea).toMatchSnapshot();
  });
});
