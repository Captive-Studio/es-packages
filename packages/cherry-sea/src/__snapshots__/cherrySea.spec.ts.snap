// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cherrySea > matches snapshot 1`] = `
{
  "color": {
    "error": "red",
    "errorContainer": "red",
    "onError": "red",
    "onErrorContainer": "red",
    "onPrimary": "#fff",
    "onPrimaryContainer": "red",
    "onSecondary": "red",
    "onSecondaryContainer": "red",
    "onSurface": "#102a43",
    "onSurfaceVariant": "#334e68",
    "onTertiary": "red",
    "onTertiaryContainer": "red",
    "outline": "red",
    "outlineVariant": "red",
    "primary": "#d9127c",
    "primaryContainer": "red",
    "secondary": "red",
    "secondaryContainer": "red",
    "surface": "#f5f7fa",
    "tertiary": "red",
    "tertiaryContainer": "red",
    "white": "#fff",
  },
  "colorPalette": {
    "aquaSerenity": {
      "100": "#D8FFFF",
      "200": "#BFFBFB",
      "300": "#A7F6F6",
      "400": "#90F0F0",
      "50": "#EFFEFE",
      "500": "#70E3E3",
      "600": "#56D2D2",
      "700": "#40C9C9",
      "800": "#38A0A0",
      "900": "#2F7777",
    },
    "blueGrey": {
      "100": "#d9e2ec",
      "200": "#bcccdc",
      "300": "#9fb3c8",
      "400": "#829ab1",
      "50": "#f5f7fa",
      "500": "#5e7997",
      "600": "#486581",
      "700": "#334e68",
      "800": "#243b53",
      "900": "#102a43",
    },
    "goldenFizz": {
      "100": "#feffc7",
      "200": "#fcff95",
      "300": "#f4fe58",
      "400": "#e4f40e",
      "50": "#fffee5",
      "500": "#c9dc06",
      "600": "#9db000",
      "700": "#758506",
      "800": "#5d690b",
      "900": "#4d5810",
    },
    "lightBlueVivid": {
      "100": "#b3ecff",
      "200": "#81defd",
      "300": "#5ed0fa",
      "400": "#40c3f7",
      "50": "#f0fbff",
      "500": "#2bb0ed",
      "600": "#188dcd",
      "700": "#1177b6",
      "800": "#0a6299",
      "900": "#035388",
    },
    "limeGreen": {
      "100": "#e2f7c2",
      "200": "#c7ea8f",
      "300": "#abdb5e",
      "400": "#94c843",
      "50": "#f2fde0",
      "500": "#7bb026",
      "600": "#63921a",
      "700": "#507712",
      "800": "#42600c",
      "900": "#2b4005",
    },
    "pinkVivid": {
      "100": "#ffb8d2",
      "200": "#ff8cba",
      "300": "#f364a2",
      "400": "#e8368f",
      "50": "#fff5f8",
      "500": "#d9127c",
      "600": "#bc0a6f",
      "700": "#a30664",
      "800": "#870557",
      "900": "#620042",
    },
    "purple": {
      "100": "#cfbcf2",
      "200": "#a081d9",
      "300": "#8662c7",
      "400": "#724bb7",
      "50": "#eae2f8",
      "500": "#653cad",
      "600": "#51279b",
      "700": "#421987",
      "800": "#34126f",
      "900": "#240754",
    },
    "redVivid": {
      "100": "#ffbdbd",
      "200": "#ff9b9b",
      "300": "#f86a6a",
      "400": "#ef4e4e",
      "50": "#fff0f0",
      "500": "#e12d39",
      "600": "#cf1124",
      "700": "#ab091e",
      "800": "#8a041a",
      "900": "#610316",
    },
    "yellowVivid": {
      "100": "#fff3c4",
      "200": "#fce588",
      "300": "#fadb5f",
      "400": "#f7c948",
      "50": "#fffbea",
      "500": "#f0b429",
      "600": "#de911d",
      "700": "#cb6e17",
      "800": "#b44d12",
      "900": "#8d2b0b",
    },
  },
  "spacing": {
    "0": "0px",
    "100": "16px",
    "1200": "192px",
    "150": "24px",
    "1600": "256px",
    "200": "32px",
    "2400": "384px",
    "25": "4px",
    "300": "48px",
    "3200": "512px",
    "400": "64px",
    "4000": "640px",
    "4800": "768px",
    "50": "8px",
    "600": "96px",
    "75": "12px",
    "800": "128px",
  },
  "typeface": {
    "brand": [
      "Barlow",
      "-apple-system",
      "Segoe UI",
      "roboto",
      "noto sans",
      "ubuntu",
      "cantarell",
      "helvetica neue",
    ],
    "plain": [
      "Inter",
      "-apple-system",
      "Segoe UI",
      "roboto",
      "noto sans",
      "ubuntu",
      "cantarell",
      "helvetica neue",
    ],
  },
}
`;
